CONTRIBUTING.md

```diff
@@ -0,0 +1,57 @@
+# Developer guidelines
+
+<!-- vim-markdown-toc GFM -->
+
+- [Development](#development)
+  - [Dev environment set up](#dev-environment-set-up)
+- [Pull requests](#pull-requests)
+- [Merging to master](#merging-to-master)
+  - [Stages: Draft and Ready for review](#stages-draft-and-ready-for-review)
+
+<!-- vim-markdown-toc -->
+
+## Development
+
+### Dev environment set up
+
+```bash
+# Create an .env based on the example
+$ cp .env.example .env
+
+# Start MySQL in docker
+$ docker compose up -d
+# Wait for the db to finish starting, it takes a few seconds before it's ready...
+
+# Setup the database
+$ npm run setup
+
+# Start the app, which will be running on localhost:3000
+$ npm run dev
+
+# Visually see the database in a GUI:
+$ npm run db:studio
+```
+
+> **Note** `npm run setup` creates a local `mysql-data/` directory, and is a one time setup. That directory can be safely deleted if the database needs to be reset, or if errors occur when starting the database.
+
+## Pull requests
+
+- To avoid duplicate work, create a draft pull request.
+- Avoid cosmetic changes to unrelated files in the same commit.
+- Use a [feature branch](https://www.atlassian.com/git/tutorials/comparing-workflows) instead of the master branch.
+
+## Merging to master
+
+For maintainers: when a PR is ready to merge to master,
+
+- prefer Squash Merge for "single-commit PRs" (when the PR has only one meaningful commit).
+- prefer Merge for "multi-commit PRs" (when the PR has multiple meaningful commits).
```

Comment:
I took a look in telescope. I didn't find a contributor guide anywhere...

Reply:
Telescope contributing docs are at https://telescope.cdot.systems/docs/contributing/

Reply:
Oh thank you

Reply:
Hm, the rebase workflow described [here](https://app.slack.com/client/T71E5T1A8/C04HR4WV5P0) seems very similar to doing `Squash and merge` on a PR, from what I understand?

Reply:
Doesn't rebase workflow for telescope involve rebasing on main instead of merging main into your branch?

I think when merging into main, we should squash and merge in all cases. Even if there are multiple commits, they will be moved in the description of the main commit message

Reply:
From what I understand from the telescope docs, it basically describes rebasing your commits down to a single commit on your feature branch. Then rebasing or merging that commit onto master.

Which Im not sure if I see the benefits of. We'd lose the commit history in the PR itself.

Reply:
Yea, when you rebase commits into one, and you pick the squash option for other commits, it ends up putting those into the description of the main commit. You can still kind of see the commits that happened to build up to the main commit.

I don't think we really need the commit history of each commit in the PR assuming we have small, meaningful PRs. If a PR has two major meaningful commits, shouldn't that be two PRs instead of one?

Reply:
If we want to keep the commit history, we could simply squash and merge when merging into main. That will keep the commit history in the PR

Reply:
Take something like this PR. This is small enough to squash into a single commit and it wouldn't make a difference.

But a PR for a new feature, say an API endpoint, or a frontend page would take a lot more work. The "meaningful" commit would be the completion of the feature. But the commit history is kinda valuable in that case.

Reply:
> If we want to keep the commit history, we could simply squash and merge when merging into main. That will keep the commit history in the PR

Right that is what im proposing here.

Reply:
Yup, I agree. That makes sense. Sorry I kinda got confused

Reply:
In that case, do you propose changing line 48 to Squash and merge too?

Reply:
yeah telescope is kinda vague on this point. So it would be nice to standardize the process here.

Reply:
Hmm, idk if its worth making the distinction between bigger PRs, like a new feature, and smaller PRs like this one. What do you think?

Reply:
We can always just say "squash and merge all PRs"

Reply:
if we dont care about that distinction

Reply:
The idea I think is bigger PRs have a higher risk associated with them. Ie pushing out a new feature could break something. So having that fine granular history could be helpful in master as well. If we ever need to revert something, we can revert a single commit, not the entire feature.

Reply:
"squash and merge all PRs" makes sense to me. A new feature could break something, but if have good tests this shouldn't happen. Maybe in extremely specific cases, we could make an exception for this if it is really needed. 

But I don't see any reason to put it in the docs

Reply:
Yeah that's fair. Just thinking a head, most things we do shouldn't break stuff. But when we do encounter when a new feature PR doesn't play well with what we have currently, we can revert the entire feature. And merge it back when it's fixed.

Reply:
I think we are gonna say we will squash and merge all PRs, [here](https://github.com/Seneca-CDOT/starchart/pull/54/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R52)

Reply:
> I think we are gonna say we will squash and merge all PRs, [here](https://github.com/Seneca-CDOT/starchart/pull/54/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R52)

Could you add a screenshot of this option in the docs just so it's clear?

Reply:
done